[GStreamer][VideoCapture] GStreamerCapturer: Disable last-sample in appsink

jameshilliard opened the pull request:
#### 66fe1fcd80981a73e08b73bf88c85714a82dda12
<pre>
[GStreamer][VideoCapture] GStreamerCapturer: Disable last-sample in appsink
<a href="https://bugs.webkit.org/show_bug.cgi?id=242961">https://bugs.webkit.org/show_bug.cgi?id=242961</a>

Reviewed by Xabier Rodriguez-Calvar.

We need to set the appsink enable-last-sample property to FALSE so that sample
buffers can be returned to their pools.

* Source/WebCore/platform/mediastream/gstreamer/GStreamerCapturer.cpp:
(WebCore::GStreamerCapturer::setupPipeline):

Canonical link: <a href="https://commits.webkit.org/253033@main">https://commits.webkit.org/253033@main</a>
</pre>

ceyusa commented on Source/WebCore/platform/mediastream/gstreamer/GStreamerCapturer.cpp:
```diff
@@ -170,6 +170,7 @@ void GStreamerCapturer::setupPipeline()
     m_sink = makeElement("appsink");
 
     gst_app_sink_set_emit_signals(GST_APP_SINK(m_sink.get()), TRUE);
+    g_object_set(GST_BASE_SINK(m_sink.get()), "enable-last-sample", FALSE, nullptr);
```
niptick: There's no need to cast m_sink for g_object_set, such as happens in the line below.

jameshilliard replied:
fixed

jameshilliard committed:
[GStreamer][VideoCapture] GStreamerCapturer: Disable last-sample in appsink
https://bugs.webkit.org/show_bug.cgi?id=242961

Reviewed by Xabier Rodriguez-Calvar.

We need to set the appsink enable-last-sample property to FALSE so that sample
buffers can be returned to their pools.

* Source/WebCore/platform/mediastream/gstreamer/GStreamerCapturer.cpp:
(WebCore::GStreamerCapturer::setupPipeline):

Canonical link: https://commits.webkit.org/253033@main
